pkg/volume/csi/csi_attacher.go

```diff
@@ -166,6 +168,46 @@ func (c *csiAttacher) waitForVolumeAttachmentInternal(volumeHandle, attachID str
 	return attach.Name, nil
 }
 
+func (c *csiAttacher) waitForVolumeAttachmentWithLister(volumeHandle, attachID string, pollInterval, timeout time.Duration) error {
+	klog.V(4).Info(log("probing VolumeAttachment [id=%v]", attachID))
+
+	ticker := time.NewTicker(pollInterval)
+	timer := time.NewTimer(timeout) // TODO (vladimirvivien) investigate making this configurable
+	defer ticker.Stop()
+	defer timer.Stop()
+
+Loop:
+	for {
+		select {
+		case <-ticker.C:
+			volumeAttachment, err := c.plugin.volumeAttachmentLister.Get(attachID)
```

Comment:
You may want to move this before select{} so that super fast attachments doesnt need to wait for first timer tick :)

Reply:
Thanks for the suggestion, I like this

Reply:
The other nice thing about this approach is it removed the need to use labels. I had to use them previous to break out of the loop because `go vet` complained about unreachable code.